docs/usage/labels.md

```diff
@@ -0,0 +1,125 @@
+
+## Management
+
+The Service Manager conforms to the [API](#api) as to how labels are managed. 
+
+Labels can be attached to or detached from a resource by `PATCH`-ing the resource with a [label change object](https://github.com/Peripli/specification/blob/visibility-labels/api.md#label-change-object).
+
+# Querying
+
+Querying can be performed both on labels and resource fields.
+A valid query consist one or more criteria and a criterion cosists of a left operand, an operator and a right operand.
+
+The syntax is described below (note that **|** is a query separator and **||** is a separator for multiple values): 
+```
+<query-syntax>              ::= <criterion> OR <criterion> "|" <query-syntax>
+<criterion>                 ::= KEY [ <multivariate-criterion> OR <univariate-criterion> ]
+<multivariate-criterion>    ::= <empty-list> OR <multivariate-operator> <multiple-values>
+<empyt-list>                ::= "[]"
+<multivariate-operator>     ::= "in" OR "notin"
+<multiple-values>           ::= "[" <values> "]"
+<values>                    ::= VALUE OR VALUE "||" <values>
+<univariate-criterion>      ::= ["=" OR "!=" OR "eqornil" OR "lt" OR "gt"] VALUE
+
+KEY is a sequence of characters with length from 1 to 255 characters, not containing a query separator.
+VALUE is a sequence of characters with length from 1 to 255 characters.  
+The query separator character (|) must be escaped with a backslash (\) if it is present.
+For array values, the separator between the values in the array is ||.  
+Delimiter between the operator and its operands is exactly one whitespace: ' '
+Example:
+x in [val1||val2]|y = 5|z eqornil value with \| separator
+```
+
+## Operators
+
+* Equals (**=**):
+    - Checks whether the left operand's value and the right operand are equal
+    - Example: `platform_id = my_platform_id`
+* Not equals (**!=**):
+    - Checks whether the left operand's value and the right operand are not equal
+    - Example: `platform_id != my_platform_id`
+* Equals or nil (**eqornil**)
+    - Checks whether the left operand's value is equal to the right operand OR the left operand's value is NULL
+    - Example: `platform_id eqornil my_platform_id`
+* Greater than (**gt**):
+    - Checks whether the left operand's value is greater than the right operand. Supports only numerical values.
+    - Example: `id gt 5`
+* Less than (**lt**)
+    - Checks whether the left operand's value is less than the right operand. Supports only numerical values.
+    - Example: `id lt 5`
+* In (**in**)
+    - Checks whether the left operand's value is contained in the right operand. Works only for list values of the right operand contained in square braces.
+    - Example: `id in [5|6|7]`
+* Not in (**notin**)
+    - Checks whether the left operand's value is NOT contained in the right operand. Works only for list values of the right operand contained in square braces.
+    - Example: `id notin [1|2|3]`
+
+## Query Types
```

Comment:
Somewhere we should mention that new lines are not allowe and we could mention how querying of fields and labels with json value for right ops work

We could also include examples for simpler scenarios